File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.cpp
class Solution {
public:
    int longestOnes(vector<int>& arr, int k) {
        int start=0,maxlen=0;
        int z=0;
        for(int i=0;i<arr.size();i++){
            if(arr[i]==0){
                z++;
                while(z>k){
                    if(arr[start]==0)
                    z--;
                    start++;
                }
            }
            maxlen=max(maxlen,i-start+1);
        }
        return maxlen;
    }
};

File: 1046-max-consecutive-ones-iii/max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int z=0, len=0;
        int start=0;
        for(int i=0;i<nums.length;i++){
            if(nums[i]==0)
            z++;
            while(z>k){
                if(nums[start]==0)
                z--;
                start++;
            }
            len=Math.max(len,i-start+1);
        }
        return len;
    }
}